pjmedia/src/pjmedia/stream.c

```diff
@@ -2176,6 +2188,9 @@ static void on_rx_rtp( pjmedia_tp_cb_param *param)
     }
 
 on_return:
+    if (!update_rtcp)
+	return;
+
```

Comment:
This return does not seem to only skip RTCP update, but it also skips sending initial RTCP & RTCP-FB, so the variable name `update_rtcp` is perhaps a bit misleading.

Even perhaps we should refactor this, e.g: some cases (e.g: valid but zero payload length) still trigger RTCP update & sending RTCP/RTCP-FB, and other cases simply return?

Reply:
Discarded packet cause:
1. invalid source address 
2. sequence status bad flag (`seq_st.status.flag.bad`) is set
3. empty payload

Only empty payload (3) will need to process/update RTCP.

Reply:
Agree, I think the change can be included in this PR too (e.g: simply by introducing another goto label).

Reply:
Actually there are other scenario which packets are ignored:
1. Error decoding RTP ([ref](https://github.com/pjsip/pjproject/blob/master/pjmedia/src/pjmedia/stream.c#L1874)). This will update rtcp stats. If source address is not valid, no need to update rtcp stats.
2. RTCP data ([ref](https://github.com/pjsip/pjproject/blob/master/pjmedia/src/pjmedia/stream.c#L1882)). No changes needed
3. Channel is paused ([ref](https://github.com/pjsip/pjproject/blob/master/pjmedia/src/pjmedia/stream.c#L1888))
If source address is not valid, process/update RTCP.

Re: new goto label
There's no other processing needed, I think we can just return from the method?

Reply:
> 3. Channel is paused ([ref](https://github.com/pjsip/pjproject/blob/master/pjmedia/src/pjmedia/stream.c#L1888))
>    If source address is not valid, process/update RTCP.

Okay. Just make sure that regular RTCP RR is still sent when the stream is paused.

> 
> Re: new goto label
> There's no other processing needed, I think we can just return from the method?

Perhaps, but RTCP session may still need to be updated (via `pjmedia_rtcp_rx_rtp2`, note the `discarded` param) regardless RTP is bad/not, please check RTCP-XR specs.